ntbs-service/Models/Validations/ValidIntRangeAttribute.cs

```diff
@@ -20,14 +20,23 @@ protected override ValidationResult IsValid(object value, ValidationContext vali
             PropertyInfo property = type.GetProperty(ComparisonValue);
             if (property != null)
             {
-                object propertyValue = property.GetValue(instance);
                 int valueToValidate;
                 int maxValue;
-                if(Int32.TryParse(propertyValue.ToString(), out maxValue) && Int32.TryParse(value.ToString(), out valueToValidate)) {
-                    if(valueToValidate >= 0 && valueToValidate <= maxValue) {
+                object propertyValue = property.GetValue(instance);
+                if(propertyValue == null) {

```

Comment:
For some reason I couldn't get 

propertyValue == null ? maxValue = 0 : Int32.TryParse(propertyValue.ToString(), out maxValue);

to work to make all of these neater, any idea why this syntax is wrong? Hopefully an easy fix

Reply:
Ah, I imagine the ternary operator (`_ ? _ : _`)can only contain values and not expressions in its branches - which is also how it's actually semantically different from `if-else`. Some languages make this difference more blurry!
If there's an equivalent to `Int32.TryParse` which doesn't use an `out` parameter, then you could do something like
`maxValue = propertyValue == null ? 0 : Int32.Parse(propertyValue.ToString())`.
Current use of if-else is also absolutely fine and perhaps one worth sticking to :)
(Unless `TryParse` accepts `null`, in which case you can use `propertyValue?.ToString()` inside it. [Looks like it does](https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=netframework-4.8#System_Int32_TryParse_System_String_System_Int32__) and even returns 0 in those cases - although if we want to use this implemenation coincidence, then we should leave a comment expaling in it as it's not obvious at first!)